### PR TITLE
Removing Random jump into Loop when qantisation is enabeled.

### PR DESCRIPTION
**Pull request by daschuer**
The main problem is fixed. 
A small random jump is still there, when only one track is loaded.

### CODE REVIEW

**Comment by daschuer** [comment]
I cannot reproduce this remaining small jump in an old master. Is this a regression?

**Comment by Be-ing** [comment]
Tests are failing:
```
[ RUN      ] LoopingControlTest.BeatLoopSize_SetAndToggle
...
src/test/looping_control_test.cpp:599: Failure
Value of: m_pLoopEnabled->toBool()
  Actual: true
Expected: false
src/test/looping_control_test.cpp:600: Failure
Value of: m_pBeatLoop2Enabled->toBool()
  Actual: true
Expected: false
in ~EngineMaster() 
[  FAILED  ] LoopingControlTest.BeatLoopSize_SetAndToggle (52 ms)
[ RUN      ] LoopingControlTest.BeatLoopSize_IsSetByNumberedControl
...
src/test/looping_control_test.cpp:645: Failure
Value of: m_pBeatLoop2Enabled->toBool()
  Actual: true
Expected: false
src/test/looping_control_test.cpp:646: Failure
Value of: m_pLoopEnabled->toBool()
  Actual: true
Expected: false
in ~EngineMaster() 
[  FAILED  ] LoopingControlTest.BeatLoopSize_IsSetByNumberedControl (288 ms)
[ RUN      ] LoopingControlTest.BeatLoopSize_ValueChangeDoesNotActivateLoop
...
src/test/looping_control_test.cpp:682: Failure
Value of: m_pLoopEnabled->toBool()
  Actual: true
Expected: false
src/test/looping_control_test.cpp:684: Failure
Value of: m_pLoopEnabled->toBool()
  Actual: true
Expected: false
src/test/looping_control_test.cpp:685: Failure
Value of: m_pBeatLoop4Enabled->toBool()
  Actual: true
Expected: false
in ~EngineMaster() 
[  FAILED  ] LoopingControlTest.BeatLoopSize_ValueChangeDoesNotActivateLoop (56 ms)
[ RUN      ] LoopingControlTest.BeatLoopSize_ValueChangeResizesBeatLoop
...
src/test/looping_control_test.cpp:701: Failure
Value of: m_pLoopEnabled->toBool()
  Actual: true
Expected: false
in ~EngineMaster() 
[  FAILED  ] LoopingControlTest.BeatLoopSize_ValueChangeResizesBeatLoop (57 ms)

**Comment by Be-ing** [comment]
Whoops I think those test failures are from my recent change.

**Comment by Be-ing** [comment]
Yeah, the test failures are from my last commit. I'll take care of that.

**Comment by Be-ing** [comment]
Pressing reloop_toggle when the play position is after the loop enables the loop but does not seek to it. Is this intended?

**Comment by Be-ing** [comment]
>  A small random jump is still there, when only one track is loaded.

I think this is a [known bug](https://bugs.launchpad.net/mixxx/+bug/1663444) with quantize that doesn't have to do with looping.

**Comment by Be-ing** [comment]
Thanks for fixing this!

File: src/engine/loopingcontrol.cpp [hunk 1]
```diff
@@ -739,7 +739,9 @@ void LoopingControl::slotLoopEndPos(double pos) {
 void LoopingControl::notifySeek(double dNewPlaypos) {
     LoopSamples loopSamples = m_loopSamples.getValue();
     if (m_bLoopingEnabled) {
-        if (dNewPlaypos < loopSamples.start || dNewPlaypos > loopSamples.end) {
+        // Disable loop when we jump after it, using hot cues or waveform overview
+        // If we jump before, the loop it is kept enabled as catching loop
+        if (dNewPlaypos > loopSamples.end) {
```
Comment (Be-ing) [inline src/engine/loopingcontrol.cpp:744]: I do not understand why LoopingControl::process() doesn't jump to the loop without setting m_bReloopCatchUpcomingLoop to true here, but it works somehow.

Comment (Be-ing) [reply src/engine/loopingcontrol.cpp:744]: Oh, m_bReloopCatchUpcomingLoop is already true because I pressed reloop_toggle.

**Comment by Be-ing** [comment]
> Pressing reloop_toggle when the play position is after the loop enables the loop but does not seek to it. Is this intended?

Nevermind, I can't reproduce this issue.

**Pull request merged**